File: skyspark-java/src/main/java/com/akikanellis/skyspark/api/algorithms/SkylineAlgorithm.java
package com.akikanellis.skyspark.api.algorithms;

import org.apache.spark.api.java.JavaRDD;

import java.awt.geom.Point2D;
import java.io.Serializable;

/**
 * A skyline algorithm is an algorithm capable of calculating the skyline set of a dataset S. Current implementations
 * include {@link com.akikanellis.skyspark.api.algorithms.bitmap.Bitmap},
 * {@link com.akikanellis.skyspark.api.algorithms.bnl.BlockNestedLoop}
 * and {@link com.akikanellis.skyspark.api.algorithms.sfs.SortFilterSkyline}
 */
public interface SkylineAlgorithm extends Serializable {

    /**
     * Given an RDD of points, return the skyline set of them.
     *
     * @param points the points to give the skyline for.
     * @return the skyline set.
     */
    JavaRDD<Point2D> computeSkylinePoints(JavaRDD<Point2D> points);
}

File: skyspark-scala/src/main/scala/com/akikanellis/skyspark/scala/api/algorithms/SkylineAlgorithm.scala
package com.akikanellis.skyspark.scala.api.algorithms

import org.apache.spark.rdd.RDD

/**
  * A skyline algorithm is an algorithm capable of calculating the skyline set of a dataset S. Current implementations
  * include [[com.akikanellis.skyspark.scala.api.algorithms.bnl.BlockNestedLoop]].
  */
trait SkylineAlgorithm extends Serializable {

  /**
    * Given an RDD of points, return their skyline set.
    *
    * @param points the points to compute the skyline for.
    * @return the skyline set.
    */
  def computeSkylinePoints(points: RDD[Point]): RDD[Point]
}
